src/commands/deploy/nuxt/deploy.ts

```diff
@@ -72,16 +97,121 @@ async function deployFunctions(config: YamlProjectConfiguration, stage?: string)
+    domainName: string,
+    config: YamlProjectConfiguration,
+    stage: string,
+) {
+    const serverOrigin: CreateFrontendV2Origin = {
+        domain: {
+            id: deployedFunctions[0].id,
+            type: "function",
+        },
+        path: undefined,
+        methods: ["GET", "HEAD", "POST", "PUT", "DELETE", "OPTIONS", "PATCH"],
+        cachePolicy: "custom-function-cache",
+    };
+
+    const s3Origin: CreateFrontendV2Origin = {
+        domain: {
+            id: "frontendHosting",
+            type: "s3",
+        },
+        path: undefined,
+        methods: ["GET", "HEAD", "OPTIONS"],
+        cachePolicy: "caching-optimized",
+    };
+
+    const paths: CreateFrontendV2Path[] = [...(await computeAssetsPaths(s3Origin))];
+
+    const { domain: distributionUrl } = await createFrontendProjectV2(
+        domainName,
+        config.name,
+        config.region,
+        stage,
+        paths,
+        /* defaultPath= */ {
+            origin: serverOrigin,
+        },
+        ["nuxt"],
+    );
+
+    if (!distributionUrl.startsWith("https://") && !distributionUrl.startsWith("http://")) {
+        return `https://${distributionUrl}`;
+    }
+
+    return distributionUrl;
+}
+
+async function deployStaticAssets(config: YamlProjectConfiguration, stage: string) {
+    const getFrontendPresignedURLPromise = getFrontendPresignedURL(
+        /* subdomain= */ undefined,
+        /* projectName= */ config.name,
+        stage,
+        /* type= */ "nextjs",
+    );
+
+    const temporaryFolder = await createTemporaryFolder();
+    const archivePath = path.join(temporaryFolder, "nuxt-static.zip");
+
+    await fs.promises.mkdir(path.join(temporaryFolder, "nuxt-static"));
+    await fs.promises.cp(
+        path.join(process.cwd(), ".output", "public"),
+        path.join(temporaryFolder, "nuxt-static"),
+        { recursive: true },
+    );
+
+    const { presignedURL, userId, domain } = await getFrontendPresignedURLPromise;
+    debugLogger.debug(`Generated presigned URL for Next.js static files. Domain: ${domain}`);
+
+    await zipDirectoryToDestinationPath(
+        path.join(temporaryFolder, "nuxt-static"),
+        domain,
+        archivePath,
+    );
+
+    await uploadContentToS3(presignedURL, archivePath, undefined, userId);
+    debugLogger.debug("Uploaded Nuxt static files to S3.");
+
+    return domain;
+}
+
+async function computeAssetsPaths(
+    s3Origin: CreateFrontendV2Origin,
+): Promise<CreateFrontendV2Path[]> {
+    const folder = path.join(process.cwd(), ".output", "public");
+    return new Promise((resolve, reject) => {
+        glob(
+            "*",
+            {
+                dot: true,
+                cwd: folder,
+            },
+            (err, files) => {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+
+                const paths: CreateFrontendV2Path[] = files.map((file) => ({
```

Comment:
What if there are more than 250 files? 😥

Reply:
We cry. I'll add a task for this.